File: framework/test2/floating/single/Floating.java
import soda.unittest.*;

import java.util.*;
import java.util.function.Supplier;
import java.util.stream.*;

import soda.leetcode.*;
import soda.unittest.Validators;
import soda.unittest.work.TestWork;

import static soda.unittest.LoggerHelper.logger;

class Solution {
    public double divide(double a, double b) {
        return a / b;
    }
}

public class Floating implements Supplier<TestWork> {

    @Override
    public TestWork get() {
        var work = new TestWork(new Solution(), "divide");
        // var work = TestWork.forStruct(Struct.class);
        // work.setValidator((e, r) -> {...});
        // work.setCompareSerial(true);
        return work;
    }

    public static void main(String[] args) throws Exception {
        new Floating().get().run();
    }

}

File: framework/test2/floating/single/Floating.scala
import soda.scala.unittest._

import scala.collection.mutable
import scala.reflect.runtime.universe.typeOf

object Solution {
  def divide(a: Double, b: Double): Double = a / b
}

class Solution {}

class Floating extends (String => String) {
  override def apply(text: String): String = {
    val work = TestWork.forObject(typeOf[Solution], "divide")
    // val work = TestWork.forStruct(...)
    // work.setValidator((R, R) => Boolean)
    // work.compareSerial = true
    work.run(text)
  }
}

object Floating extends App {
  println(new Floating()(Utils.fromStdin()))
}
